spec/osut_tests_spec.rb

```diff
@@ -1333,1134 +1129,2447 @@ module M
+    expect(ctl.solarDistribution).to eq("FullInteriorAndExterior")
+    expect(ctl.isSolarDistributionDefaulted).to be(false)
+    expect(ctl.shadowCalculation.get).to eq(shd)
+
+    file = File.join(__dir__, "files/osms/out/mini.osm")
+    model.save(file, true)
+    # Simulation results E+ 22.2 (SutherlandHodgman, PolygonClipping)
+    # ... no E+ errors/warnings.
+    #
+    # office.osm  FullInteriorAndExterior                 247 GJ 3.8 UMH cool
+    # mini.osm A  FullInteriorAndExterior                 247 GJ 4.0 UMH cool
+    # mini.osm B  FullExteriorWithReflections             248 GJ 3.8 UMH cool
+    # mini.osm C  FullInteriorAndExteriorWithReflections  247 GJ 3.8 UMH cool
+
+    # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+    # Mini as an interior courtyard:
+    #   - 4x new outdoor-facing walls (core)
+    #   - remove mini floor @Z0 and mini ceiling @Z3.05
+    #   - attic mini floor facing outdoors
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[0]
+    vtx << mini_floor_vtx[3]
+    vtx << mini_floor_vtx[0]
+    vtx << mini_ceiling_vtx[3]
+    mini_north = OpenStudio::Model::Surface.new(vtx, model)
+    mini_north.setName("Mini north")
+    expect(mini_north.setSpace(core)).to be(true)
+    expect(mini_north.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[3]
+    vtx << mini_floor_vtx[0]
+    vtx << mini_floor_vtx[1]
+    vtx << mini_ceiling_vtx[2]
+    mini_east = OpenStudio::Model::Surface.new(vtx, model)
+    mini_east.setName("Mini east")
+    expect(mini_east.setSpace(core)).to be(true)
+    expect(mini_east.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[2]
+    vtx << mini_floor_vtx[1]
+    vtx << mini_floor_vtx[2]
+    vtx << mini_ceiling_vtx[1]
+    mini_south = OpenStudio::Model::Surface.new(vtx, model)
+    mini_south.setName("Mini south")
+    expect(mini_south.setSpace(core)).to be(true)
+    expect(mini_south.outsideBoundaryCondition).to eq("Outdoors")
+
+    vtx = OpenStudio::Point3dVector.new
+    vtx << mini_ceiling_vtx[1]
+    vtx << mini_floor_vtx[2]
+    vtx << mini_floor_vtx[3]
+    vtx << mini_ceiling_vtx[0]
+    mini_west = OpenStudio::Model::Surface.new(vtx, model)
+    mini_west.setName("Mini west")
+    expect(mini_west.setSpace(core)).to be(true)
+    expect(mini_west.outsideBoundaryCondition).to eq("Outdoors")
+
+    mini_floor.remove
+    mini_ceiling.remove
+    expect(mini_attic.setOutsideBoundaryCondition("Outdoors")).to be(true)
+
+    # Re-validating pre-tested areas + volumes.
+    expect(core.floorArea).to be_within(TOL).of(149.66 - 4) # -mini m2
+    core_volume = core.floorArea * core.ceilingHeight
+    expect(core_volume).to be_within(TOL).of(core.volume)
+    expect(attic.volume).to be_within(TOL).of(720.19)
+    expect(attic.floorArea).to be_within(TOL).of(567.98)
```

Comment:
Adding a **mini** (exterior) courtyard is correctly reflected in area and volume calculations for both core and attic spaces.